src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+    @isTest
+    public static void testDisabledAddr() {        
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Contacts_Addresses_Enabled__c = false));
+
+        List<Contact> contacts = UTIL_UnitTestData_TEST.CreateMultipleTestContacts(2);
+        for (Contact contact : contacts) {
+            contact.MailingStreet = 'new street';
+            contact.MailingCity = 'new city';
+        }
+
+        // verify Contact address
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertEquals(null, contact.Current_Address__c);
+        }
+        
+        // verify no address objects created
+        List<Address__c> ListAddr = [select Id from Address__c];
+        system.assertEquals(0, ListAddr.size());
+    }
+    
+    /*********************************************************************************************************
+    @description Update Contacts' Mailing address fields to empty cause clearing the default address  
+    verify:
+        existing Default Address updated to be non-default
+        no new Address created
+    **********************************************************************************************************/            
+    static testMethod void clearAddrExisting() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Contacts_Addresses_Enabled__c = true));
+                
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(4);
+
+        // clear addresses fields directly
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            contactsAddrs.contacts[i].MailingStreet = null;
+            contactsAddrs.contacts[i].MailingCity = null;
+            contactsAddrs.contacts[i].MailingState = null;
+            contactsAddrs.contacts[i].MailingPostalCode = null;
+            contactsAddrs.contacts[i].MailingCountry = null;
+            if (ADDR_Addresses_UTIL.isStateCountryPickListsEnabled) {
+                contactsAddrs.contacts[i].put('MailingStateCode', null);
+                contactsAddrs.contacts[i].put('MailingCountryCode', null);
+            }             
+        }
+
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        update contactsAddrs.contacts;
+        Test.stopTest();
+        
+        // verify that no new address was created
+        List<Address__c> addrs = [select Default_Address__c from Address__c];
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.